docs/interactions.rst

```diff
@@ -0,0 +1,131 @@
+
+As shown in the demonstration below you make a main slash command or a dummy slash command and build sub-commands off it.
+
+.. code-block:: python3
+
+        @bot.slash_command(guild_ids=[...])  # Making the command and limiting the guilds
+        async def main(interaction):  
+        # passing through interaction and indentifing the sub-command group name
+        await interaction.response.send_message(
+        "This will never get called if this has subcommands."
+            )  # a function never called
+
+
+        @main.subcommand()  # Identifying The Sub-Command
+        async def sub1(interaction):  # Making The Sub Command Name And Passing Through Interaction
+        await interaction.response.send_message(
+            "This is subcommand 1!"
+            )  # Sending A Response
+
+
+        @main.subcommand(description="Aha! Another subcommand")  
+        # Identifying The Sub-Command And Adding A Descripton
+        async def subcommand_two(interaction: Interaction):  # Passing in interaction
+
+        await interaction.response.send_message(
+            "This is subcommand 2!"
+            )       # Responding With The Args/Fields
+        
+Fields And Requirements
+-----------------------
+Fields are meant to facilitate an easier way to fill info, letting people using your slash command get a different response for each answer.
+
+Nextcord's implementation of slash commands has fields and is very simple. in the example below is a field.
+
+.. code-block:: python3
+     
+     @bot.slash_command(name="help")
+     async def help(
+         interaction: Interaction,
+         setting: str = SlashOption(name="settings", description="Configure Your Settings")
+     ):
+         if setting == "music":
+             await interaction.response.send_message(f"Sorry we aren't connected to lavalink currently")
+         elif setting == "moderation":
+             await interaction.response.send_message(f"Moderation?")
+         else:
+             await interaction.response.send_message("Odd, I don't know that setting")
+
+
+How To Make Slash Commands In Cogs
+----------------------------------
+Show below is an example of a simple command running in a cog.
+
+.. code-block:: python3
+
+    class ExampleCog(commands.Cog):
+        def __init__(self):
+            self.count = 0
+
+        @slash_command(name="cogexample")
+        async def slash_example_cog_command(self, interaction):
+            await interaction.response.send_message("Hello i am a slash command in a cog!")
+
+The example shown above responds to a user when they do a slash command. It is identical to a normal slash command and to normal commands in general.
+
+How To Make Context Menu Commands
+---------------------------------
+Context Menu commands display commands on a menu of a message/user.
+
+User Commands
+~~~~~~~~~~~~~~
+What you see below is a example of a simple user command.
+It is a user dump command that dumps user data.
```

Comment:
This could probably be simplified a bit and like just give the users username? Also possibly say something about thats the command when you use the apps menu on a user?